src/components/SDDialog.react.js

```diff
@@ -14,7 +14,7 @@ type Props = {
   children?: Node,
   modal?: boolean,
   open?: boolean,
-  setProps?: (props: { modal: boolean, open: boolean }) => void,
+  setProps?: () => void,
```

Comment:
This should be more explicit about what can be passed into `setProps` as an argument.

Example: `setProps?: (props: { modal?: boolean, open?: boolean }) => void`